Update create function （file-upload verup）

FUM01ichikawa opened the pull request:
# what
・出品機能に投稿する画像を表示させる
実装内容
ー10枚までアップロード可能
ー画像を削除する

未実装※今後実装したい
・投稿した内容→商品詳細ページでの画像表示の順番を入れ替える
（現在は最後に投稿されたものが一番最初に表示される）
・保存名称の調整（nameカラムに入る情報が、ファイル名とは異なる）
・画像の編集ボタン→編集機能
・ドラッグ＆ドロップでの画像追加

# why
・必須機能の中で重要な機能のため、追加を行う

KassemberKiyoshi committed:
Merge branch 'master' into Update1-create-function

KassemberKiyoshi committed:
Merge branch 'master' into Update1-create-function

KassemberKiyoshi committed:
Update-create-function(image_form ver-up)

# what
・出品機能に投稿する画像を表示させる
実装内容
ー10枚までアップロード可能
ー画像を削除する

未実装
・投稿した内容→商品詳細ページでの画像表示の順番を入れ替える
（現在は最後に投稿されたものが一番最初に表示される）
・保存名称の調整（nameカラムに入る情報が、ファイル名とは異なる）
・画像の編集ボタン→編集機能

# why
・必須機能の中で重要な機能のため、追加を行う

wheesnoza commented:
LGTM!

FUM01ichikawa commented:
画像追加１
https://gyazo.com/641f6c16f929649cf6c22a12314c5fe0
画像追加２
https://gyazo.com/5395ac51b597a02af8212d80820a3f72
削除動作
https://gyazo.com/339c19664900b65c6ca169e079909d8b
投稿→商品確認
https://gyazo.com/2b803cae099995eaf3c38d2de95556e7

TOMARUMARU commented on app/controllers/products_controller.rb:
```diff
@@ -10,18 +10,22 @@ def new
   end
 
   def create
     new_product = current_user.products.create(product_params)
```
createしちゃうとエラーハンドリングの意味がないですね…

FUM01ichikawa replied:
ありがとうございます。ご指摘の点修正いたしました。

KassemberKiyoshi committed:
ご指摘の点修正

ご指摘の点修正しました。
new_product = current_user.products.create(product_params)
↓
new_product = current_user.products.new(product_params)

また、チームメンバーがreversを使って順番を入れ替えてくれました！
合わせて確認お願いします。

miyagit commented on app/controllers/products_controller.rb:
```diff
@@ -10,18 +10,24 @@ def new
+    new_product = current_user.products.new(product_params)
     if new_product.save
+      images_form = params[:images][:image_form].reverse
+      binding.pry
```
binding.pryはPRには削除しましょう。

FUM01ichikawa replied:
削除しました！

KassemberKiyoshi committed:
Binding.pry削除

miyagit commented on app/views/root/_item-card.html.haml:
```diff
@@ -2,7 +2,8 @@
 
     .item-card-photo.sold
-      = image_tag product.images.first.name, class: "lazyloaded"
+      - if product.images.first      
```
こちらってもしimageがあるならってことでしょうか？

FUM01ichikawa replied:
そうです！

shingoteshima commented:
LGTMです！
